jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java

```diff
@@ -873,7 +873,7 @@ public Resource getResource(String pathInContext) throws MalformedURLException
             // addPath with accept non-canonical paths that don't go above the root,
             // but will treat them as aliases. So unless allowed by an AliasChecker
             // they will be rejected below.
-            Resource resource = baseResource.addPath(pathInContext);
+            Resource resource = baseResource.getResource(pathInContext);
```

Comment:
I checked and this is using Path APIs for the add rather than URI apis.

Reply:
This needs to be an efficient version of:
```suggestion
            Resource resource = Resource.newResource(_factory.getResource("/").getURI().resolve(pathInContext));
```

Actually, this is still wrong, because if the base resource is a ResourceCollection then the `getURI()` will collapse the "wave function" to a single URI.

I think we need a new API on `Resource` to do a URI resolve.... in fact why not call it `Resource.resolve(String)` and have it do the URI magic internally.
For a ResourceCollection, it will need to have an internal private list of URIs for each base, which it can resovlve against, convert to a Path and check for existence.  If it is a directory, return a new ResourceCollection, else return a PathResource